blockchain/chain.go

```diff
@@ -1024,7 +1027,7 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 		// checkConnectBlock gets skipped, we still need to update the UTXO
 		// view.
 		if b.index.NodeStatus(n).KnownValid() {
-			err = view.fetchInputUtxos(b.db, nil, block)
+			err = view.fetchInputUtxos(nil, b.utxoCache, block)
```

Comment:
With this change, is it correct that now `fetchInputUtxos` will never be called with a `nil` second argument, so the extra logic to account for that case can be removed?

Reply:
The current code has it so that it still fetches from the database directly on disconnects but this isn't needed and causes more problems.

We can get rid of the option for calling `fetchInputUtxos` with the database.

Reply:
Addressed in the latest push